code-challenges.test.js

```diff
@@ -17,51 +17,135 @@
 
 // a) Create a test with an expect statement using the variable provided.
 
+describe("when codedMessage is called", () => {
+    it ("returns a coded message that converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0", () => {
+        
+        var secretCodeWord1 = "lackadaisical"
+        var secretCodeWord2 = "gobbledygook"
+
+        expect(codedMessage(secretCodeWord1)).toEqual("l4ck4d41s1c4l")
+        expect(codedMessage(secretCodeWord2)).toEqual("g0bbl3dyg00k")
+    })
+})
+
 var secretCodeWord1 = "lackadaisical"
 // Expected output: "l4ck4d41s1c4l"
 var secretCodeWord2 = "gobbledygook"
 // Expected output: "g0bbl3dyg00k"
 
-
-
-
-
 // b) Create the function that makes the test pass.
 
-
-
+// create a function that takes a string as an argument
+const codedMessage = (str) => {
+    //convert a to 4, e to 3, i to 1, and o to 0
+    // created an object to store the variables to change and their new values
+    var mapObj = {
+        a:4,
+        e:3, 
+        i:1,
+        o:0
+    }
+    // function that finds the values and replaces with the matched values.
+    strNew = str.replace(/a|e|i|o/gi, (matched) => {
```

Comment:
it's possible to do the iteration on the string without having to redeclare the letters in the regex. Consider using `.map()`